docs/options api.md

```diff
@@ -0,0 +1,204 @@
+```python
+if self.multiworld.starting_sword[self.player]:
+    do_some_things()
+```
+
+or if I need a boolean object, such as in my slot_data I can access it as:
+```python
+start_with_sword = bool(self.multiworld.starting_sword[self.player].value)
+```
+
+## Generic Option Classes
+These options are generically available to every game automatically, but can
+be overridden for slightly different behavior, if desired. See
+`worlds.soe.options.py` for an example.
+
+### Accessibility
+Sets rules for availability of locations for the player. Items is for all items
+available but not necessarily all locations, such as self-locking keys, but
+needs to be done by world for this to be different from locations access.
+
+### ProgressionBalancing
+Algorithm for moving progression items into earlier spheres to make the gameplay
+experience a bit smoother. Can be overridden if you want a different default value.
+
+### LocalItems
+Forces the players' items local to their world.
+
+### NonLocalItems
+Forces the players' items outside their world.
+
+### StartInventory
+Allows the player to define a dictionary of starting items with item name and
+quantity.
+
+### StartHints
+Gives the player starting hints for where the items defined here are.
+
+### StartLocationHints
+Gives the player starting hints for the items on locations defined here.
+
+### ExcludeLocations
+Marks locations given here as `LocationProgressType.Excluded` so that progression
+items can't be placed on them.
+
+### PriorityLocations
+Marks locations given here as `LocationProgressType.Priority` forcing progression
+items on them.
+
+### ItemLinks
+Allows users to share their item pool with other players. Currently item links
+are per game. A link of one game between two players will combine their items
+in the link into a single item, which then gets replaced with `World.create_filler()`.
+
+## Basic Option Classes
+### Toggle
+The example above. This simply has 0 and 1 as its available results with 0
+(false) being the default value. Cannot be compared to strings but can be
+directly compared to True and False.
+
+### DefaultOnToggle
+Like toggle but 1 (true) is the default value.
+
+### Choice
+A numeric option allowing you to define different sub options. Values are stored
+as integers but you can also do comparison methods with the class and strings so
+if you have an `option_early_sword` this can be compared with:
+```python
+if self.multiworld.sword_availability[self.player] == "early_sword":
+    do_early_sword_things()
+```
+
+or:
+```python
+from .options import SwordAvailability
+if self.multiworld.sword_availability[self.player] == SwordAvailability.option_early_sword:
+    do_early_sword_things()
+```
+
+### Range
+A numeric option allowing a variety of integers including the endpoints. Has a
+default `range_start` of 0 and default `range_end` of 1. Allows for negative
+values as well. This will always be an integer and has no methods for string
+comparisons.
```

Comment:
```suggestion
A numeric option allowing a variety of integers including the endpoints. Has a
default `range_start` of 0 and default `range_end` of 1. Allows for negative
values as well. This will always be an integer, and cannot be compared to a
string.
```